### PR TITLE
optimize code in RouteTable

### PR DESCRIPTION
**Pull request by kang-hl**
### Motivation:

Explain the context, and why you're making that change.
To make others understand what is the problem you're trying to solve.

### Modification:

Describe the idea and modifications you've done.

### Result:

Fixes #<GitHub issue number>.

If there is no issue then describe the changes introduced by this PR.

### CODE REVIEW

**Review by fengjiachun — APPROVED** [review]
LGTM

File: jraft-core/src/main/java/com/alipay/sofa/jraft/RouteTable.java [hunk 1]
```diff
@@ -86,11 +86,7 @@ public boolean updateConfiguration(final String groupId, final Configuration con
     private GroupConf getOrCreateGroupConf(final String groupId) {
         GroupConf gc = this.groupConfTable.get(groupId);
         if (gc == null) {
-            gc = new GroupConf();
-            final GroupConf old = this.groupConfTable.putIfAbsent(groupId, gc);
-            if (old != null) {
-                gc = old;
-            }
+            return this.groupConfTable.computeIfAbsent(groupId, key -> new GroupConf());
```
Comment (fengjiachun) [inline jraft-core/src/main/java/com/alipay/sofa/jraft/RouteTable.java:89]: Can we only keep this line as is and delete everything else in this method?

**Review by fengjiachun — CHANGES_REQUESTED** [review]

File: jraft-core/src/main/java/com/alipay/sofa/jraft/RouteTable.java (hunk 1 above)
Comment (kang-hl) [reply jraft-core/src/main/java/com/alipay/sofa/jraft/RouteTable.java:89]: > Can we only keep this line as is and delete everything else in this method?

https://bugs.openjdk.java.net/browse/JDK-8161372

**Comment by fengjiachun** [comment]
@kang-hl Thank you

**Pull request merged**